File: UnderSociety/src/main/java/undersociety/models/Message.java
package undersociety.models;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.JoinColumn;
import javax.persistence.OneToOne;
import javax.persistence.Table;

@Entity
@Table(name = "message")
public class Message  implements Comparable<Message>{
	
	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	private int idmessage;
	@OneToOne
	@JoinColumn(name = "iduser", referencedColumnName = "idusers")
	private Users iduser;
	@OneToOne
	@JoinColumn(name = "iduserto", referencedColumnName = "idusers")
	private Users iduserto;
	@Column
	private String message;
	@Column
	private String time;
	
	
	
	public Message() {}
	public Message(Users iduser, Users iduserto, String message, String time) {
		super();
		this.iduser = iduser;
		this.iduserto = iduserto;
		this.message = message;
		this.time = time;
	}
	public int getIdmessage() {
		return idmessage;
	}
	public void setIdmessage(int idmessage) {
		this.idmessage = idmessage;
	}
	public Users getIduser() {
		return iduser;
	}
	public void setIduser(Users iduser) {
		this.iduser = iduser;
	}
	public Users getIduserto() {
		return iduserto;
	}
	public void setIduserto(Users iduserto) {
		this.iduserto = iduserto;
	}
	public String getMessage() {
		return message;
	}
	public void setMessage(String message) {
		this.message = message;
	}
	
	public String getTime() {
		return time;
	}
	public void setTime(String time) {
		this.time = time;
	}
	@Override
	public int compareTo(Message u) {
		if (getIdmessage() == 0 || u.getIdmessage() == 0) {
		      return 0;
		    }
		    return (getIdmessage() - u.getIdmessage());
	}
	
	
}

File: Frontend/src/app/Class/Messages/message.ts
import { Users } from "../Users/users";

export class Message {
    private idmessage : Number;
    private iduser : Users;
    private iduserto : Users;
    private message : String;
    private time : String;

	constructor( idmessage: Number,  iduser: Users,  iduserto: Users,  message: String,  time: String) {
		this.idmessage =  idmessage;
		this.iduser =  iduser;
		this.iduserto =  iduserto;
		this.message =  message;
		this.time =  time;
	}

	public getIdmessage(): Number {
		return this.idmessage;
	}

	public getIduser(): Users {
		return this.iduser;
	}

	public getIduserto(): Users {
		return this.iduserto;
	}

	public getMessage(): String {
		return this.message;
	}

	public getTime(): String {
		return this.time;
	}

	public setIdmessage(value: Number) {
		this.idmessage = value;
	}

	public setIduser(value: Users) {
		this.iduser = value;
	}

	public setIduserto(value: Users) {
		this.iduserto = value;
	}

	public setMessage(value: String) {
		this.message = value;
	}

	public setTime(value: String) {
		this.time = value;
	}
}
